aws-api/src/test/java/com/amplifyframework/api/aws/GsonGraphQLResponseFactoryTest.java

```diff
@@ -122,15 +126,94 @@ public void partialResponseRendersWithTodoDataAndErrors() throws ApiException {
+     * merged into extensions)
+     *
+     * @throws ApiException From API configuration
+     */
+    @Test
+    public void errorResponseDeserializesExtensionsMap() throws ApiException {
+        // Arrange some JSON string from a "server"
+        final String partialResponseJson =
+                Resources.readAsString("error-extensions-gql-response.json");
+
+        // Act! Parse it into a model.
+        final GraphQLResponse<ListTodosResult> response =
+                responseFactory.buildSingleItemResponse(partialResponseJson, ListTodosResult.class);
+
+        // Assert that the response contained things...
+        String message = "Conflict resolver rejects mutation.";
+        List<GraphQLLocation> locations = Arrays.asList(
+                new GraphQLLocation(11, 3));
+        List<GraphQLPathSegment> path = Arrays.asList(
+                new GraphQLPathSegment("listTodos"),
+                new GraphQLPathSegment("items"),
+                new GraphQLPathSegment(0),
+                new GraphQLPathSegment("name")
         );
 
-        final List<GraphQLResponse.Error> actualErrors = response.getErrors();
+        Map<String, Object> data = new LinkedTreeMap<>();
+        data.put("id", "EF48518C-92EB-4F7A-A64E-D1B9325205CF");
+        data.put("title", "new3");
+        data.put("content", "Original content from DataStoreEndToEndTests at 2020-03-26 21:55:47 " +
+                "+0000");
+        data.put("_version", 2.0);
+
+        Map<String, Object> extensions = new LinkedTreeMap<>();
+        extensions.put("errorType", "ConflictUnhandled");
+        extensions.put("errorInfo", null);
+        extensions.put("data", data);
+
+        GraphQLResponse.Error expectedError = new GraphQLResponse.Error(message, locations, path, extensions);
+        GraphQLResponse<ListTodosResult> expectedResponse = new GraphQLResponse<>(null,
+                Arrays.asList(expectedError, expectedError, expectedError, expectedError));
+
+        assertNotNull(response);
+        assertNotNull(response.getErrors());
+        assertEquals(expectedResponse.getErrors().size(), response.getErrors().size());
+
+        // Assert that each error has been parsed to the same output object.
+        for (int i = 0; i < response.getErrors().size(); i++) {
+            GraphQLResponse.Error actual = response.getErrors().get(i);
+            GraphQLResponse.Error expected = expectedResponse.getErrors().get(i);
+            assertEquals("Unexpected message on error[" + i + "]", expected.getMessage(), actual.getMessage());
+            assertEquals("Unexpected extensions on error[" + i + "]", expected.getExtensions(), actual.getExtensions());
+            assertEquals("Unexpected path on error[" + i + "]", expected.getPath(), actual.getPath());
+            assertEquals("Unexpected locations on error[" + i + "]", expected.getLocations(), actual.getLocations());
+            assertEquals("Unexpected error[" + i + "]", expected, actual);
```

Comment:
Assuming that `GraphQLResponse.Error`'s `equals()` implementation considers all of `message`, `extensions`, `path`, `locations`, you should _only_ need L211, and not L207-210.

Reply:
Agreed.  I had left these here because it makes it easier to debug when this test fails.  That said, adding toString() implementations to each class should solve that problem, so I'll go ahead and remove them!